metricbeat/module/kafka/consumergroup/consumergroup_integration_test.go

```diff
@@ -0,0 +1,91 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package consumergroup
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"testing"
+	"time"
+
+	saramacluster "github.com/bsm/sarama-cluster"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/tests/compose"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+)
+
+const (
+	kafkaDefaultHost = "localhost"
+	kafkaDefaultPort = "9092"
+)
+
+func TestData(t *testing.T) {
+	compose.EnsureUp(t, "kafka")
+
+	c, err := startConsumer(t, "metricbeat-test")
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "starting kafka consumer"))
+	}
+	defer c.Close()
+
+	ms := mbtest.NewReportingMetricSetV2(t, getConfig())
+	for retries := 0; retries < 3; retries++ {
+		err = mbtest.WriteEventsReporterV2(ms, t, "")
+		if err == nil {
+			return
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+	t.Fatal("write", err)
+}
+
+func startConsumer(t *testing.T, topic string) (io.Closer, error) {
+	brokers := []string{getTestKafkaHost()}
+	topics := []string{topic}
+	return saramacluster.NewConsumer(brokers, "test-group", topics, nil)
```

Comment:
I started trying to create and joining a consumer group using `sarama` only, but I ended copying parts of `sarama-cluster`, so even if it is only for tests I think it worths to add this library.

Reply:
Number of consumers in a consumer-group are limited by the number of partitions in a topic. If we pre-initialize the topics with a known set of partitions, then we just have to setup N go-routines joining a consumer group. Using saramacluster might be the easier option though :)